src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -191,7 +194,15 @@ private static bool ShouldAnalyze(IMethodSymbol methodSymbol, WellKnownTypeProvi
                 return false;
             }
 
-            if (IsSkippedMethod(methodSymbol, wellKnownTypeProvider))
+            // Do not analyze constructors and finalizers.
+            if (methodSymbol.IsConstructor() || methodSymbol.IsFinalizer())
+            {
+                return false;
+            }
+
+            // Do not analyze public APIs for web projects
+            // See https://github.com/dotnet/roslyn-analyzers/issues/3835 for details.
+            if (isWebProject && methodSymbol.IsExternallyVisible())
```

Comment:
Shall we really skip `protected` members? I don't know exactly if they expect the member to be `public` for the reflection or if `protected` is ok.

Reply:
I am not sure, but I made is as conservative as possible, given protected members in public types are public APIs.

Reply:
It seems that `Application_Start` could be `protected` so it's mandatory anyways.